packages/databricks-vscode/src/configuration/auth/MetadataService.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import * as http from "node:http";
+import * as crypto from "node:crypto";
+import {Disposable} from "vscode";
+import {AddressInfo} from "node:net";
+import {ApiClient} from "@databricks/databricks-sdk";
+
+export class MetadataService implements Disposable {
```

Comment:
Do we have 1 metadata service per vscode process or per databricks project?

Reply:
We have one service per extension activation. AFAIK this corresponds to a VSCode window.